### PR TITLE
feat: add json schema for manifest

### PR DESCRIPTION
**Pull request by melotic**
Adds a JSON Schema for the Manifest format. Generated with [`JSON.NET Schema`](https://www.newtonsoft.com/jsonschema/help/html/Introduction.htm)

```csharp
using Microsoft.ComponentDetection.Contracts.BcdeModels;
using Newtonsoft.Json.Schema.Generation;

var generator = new JSchemaGenerator();
var schema = generator.Generate(typeof(ScanResult));

// Add meta schema
schema.ExtensionData.Add("$schema", "http://json-schema.org/draft-07/schema#");

Console.WriteLine(schema.ToString());
```

Part of #570

### CODE REVIEW

**Comment by JamieMagee** [comment]
Do you intend this to be manually or automatically maintained? Cause that will affect how I review this.

**Comment by melotic** [comment]
> Reference in new issue

I think as part of CI we should run that small C# program and diff the output. If the JSON schema changed, fail the build? Sort of like how ADO fails if the generated SQL stuff isn't part of the commit.

File: docs/schema/manifest.schema.json [hunk 1]
```diff
@@ -0,0 +1,419 @@
+{
+    "definitions": {
```
Comment (JamieMagee) [inline docs/schema/manifest.schema.json:2]: This is missing a meta-schema. It needs a `$schema` property set to `http://json-schema.org/draft-07/schema#` or `http://json-schema.org/draft-07/schema#` or some other version

**Comment by melotic** [comment]
Should I include the script and CI to check that the schema is properly updated? I'll follow up with a PR to update our validation tests to also ensure the schema is followed.

**Comment by JamieMagee** [comment]
> Should I include the script and CI to check that the schema is properly updated? I'll follow up with a PR to update our validation tests to also ensure the schema is followed.

I'll approve this now, and feel free to followup in another PR, or open an issue in our backlog.

**Review by JamieMagee — APPROVED** [review]

**Pull request merged**